src/modules/verifiable-credentials/verifiable-credentials.types.ts

```diff
@@ -0,0 +1,97 @@
+import { TypedDataField } from '@ethersproject/abstract-signer';
+import {
+  ICredentialSubject,
+  IJsonLdCredential,
+  IPresentation,
+  IPresentationDefinition,
+} from '@sphereon/pex';
+
+export interface IssuerFields {
+  key: string;
+  value: string | number;
+}
+export interface CreateCredentialParams {
+  subject: string;
+  namespace: string;
+  version: string;
+  issuerFields?: IssuerFields[];
+}
+export interface Credential<T extends ICredentialSubject>
+  extends IJsonLdCredential {
+  credentialSubject: T;
+}
+
+export interface EWFCredentialSubject extends ICredentialSubject {
+  role: {
+    namespace: string;
+    version: string;
+  };
+  issuerFields: IssuerFields[];
+  id: string;
+}
+
+export interface VerifiableCredential<T extends ICredentialSubject>
```

Comment:
@Harasz Do you think, if `EWCredentialSubject` could be used in place of `ICredentialSubject`? I think the prior will suffice our requirements.

Reply:
Definitely, but if we have a `ICredentialSubject` it's more generic.